client/src/partials/patients/registry/registry.js

```diff
@@ -14,60 +14,68 @@ PatientRegistryController.$inject = [
  */
 function PatientRegistryController($translate, Patients, $uibModal) {
   var vm = this;
-  // options for the UI grid
   
   vm.search = search;
   vm.momentAge = momentAge;
-    
+ 
+  // track if module is making a HTTP request for patients
+  vm.loading = false;
+  
   /** TODO MANAGE COLUMN : LAST_TRANSACTION */
+  // the column attribute `displayName` must be used in favour of `name` in order to allow `headerCellFilter` to function
   vm.uiGridOptions = {
-    appScopeProvider : vm, // ensure that the controller's `this` variable is bound to appScope
+    appScopeProvider : vm,
     enableColumnMenus : false,
+    headerCellFilter: 'translate',
     columnDefs : [
-      { field : 'reference', name : $translate.instant('TABLE.COLUMNS.REFERENCE') },
-      { field : 'patientName', name : $translate.instant('TABLE.COLUMNS.NAME') },
-      { field : 'patientAge', name : $translate.instant('TABLE.COLUMNS.AGE') },
-      { field : 'sex', name : $translate.instant('TABLE.COLUMNS.GENDER') },
-      { field : 'hospital_no', name : $translate.instant('TABLE.COLUMNS.HOSPITAL_FILE_NR') },
-      { field : 'registration_date', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DATE_REGISTERED') },
-      { field : 'last_visit', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.LAST_VISIT') },
-      { field : 'dob', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DOB') }  
+      { field : 'reference', displayName : 'TABLE.COLUMNS.REFERENCE', headerCellFilter: 'translate' },
```

Comment:
`$translate.instant()` is simply not feasible given our current application loading process. T

To prove this you can navigate to any page using a grid like this (billing services for example) and refresh the page you will notice that `$translate.instant()` returns the translation key - this is not acceptable. 

In addition to this often `$translate.instant()` is called before the application language has been loaded from the cache, returning the default translation language instead of the users preference. 

A work around to this is to do something along the lines of 
```js
let keysArray = //...

$translate(keysArray)
  .then((columnHeadings => configureGrid() );
```

I would suggest that `$translate.instant` is not a solution in this case.